metricflow/query/query_parser.py

```diff
@@ -274,9 +282,6 @@ def _parse_and_validate_query(
             # If the time constraint is all time, just ignore and not render
             time_constraint = None
 
-        if len(group_by_names) == 0:
```

Comment:
Previously, we were only checking if there were no dimensions at all for metrics that require a time dimension. This wouldn't catch the scenario when there is a dimension in the query but it isn't a time dimension (in which case we should error).